### ISSUE TITLE
supporting browser-native scrollbar look-and-feel

### ISSUE THREAD

**Issue by PerBothner** [xtermjs/xterm.js#5159]
I missed the discussion about replacing the native scrollbar with a custom widget imported from VsCode. I'm sure the new implementation has a number of advantage, including a more consistent UI in VSCode.

However, the cost is a UI that is inconsistent if embedded within an application where other parts of the application use the native browser scrollbar.  It may be inconsistent with other applications that use a platform toolkit (assuming the browser matches that L&F). Furthermore, if the terminal has embedded HTML content (which I am working on), and that content includes a scrollbar, then we get inconsistency even within a terminal.

Bottom line: If we cannot use xterm.js with a browser-native scrollbar (or something that mimics the look-and-feel very well) that seems like a major regression.

**Comment by Tyriar** [comment]
The biggest problem with the old one is it was built on a bunch of hacks that I'm glad to be rid of, fixing some bugs in the process, giving better trackpad support and greatly reducing the maintenance burden. We also now have the overview ruler which requires a transparent scroll bar in order to work. What ended up triggering me to finally do it was Chromium seemed to have broken the CSS rules we used to style it similar to Monaco Editor.

I don't think looking back is productive here tbh, especially since one of the features is inherently incompatible with the native opaque scroll bars. If it's very important to get something similar to before, you could either try style the new scroll bar to match native styles or fork and bring back the old `Viewport.ts` and try adapt it to the new one's interface to reduce uplevel conflicts. Though as I mentioned it's quite flawed and hacky.

> Bottom line: If we cannot use xterm.js with a browser-native scrollbar (or something that mimics the look-and-feel very well) that seems like a major regression.

FWIW this will be shipped in v6, along with some other breaking changes like the canvas renderer going away.
